### PR TITLE
tidb: update tidb_auto_analyze_ratio range

### PR DESCRIPTION
**Pull request by hi-rustin**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Updated the range of `tidb_auto_analyze_ratio`.

We don't allow users to set it to 0 anymore.

See https://github.com/pingcap/tidb/pull/52190

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by songrijie — APPROVED** [review]
It makes sense that we will not take ratio higher than one in the future version, which makes confusion.  The old setting will be taken care as what it does in the past. This looks good to me.

File: system-variables.md [hunk 1]
```diff
@@ -1072,7 +1072,7 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
 - 类型：浮点数
 - 默认值：`0.5`
-- 范围：`[0, 18446744073709551615]`
+- 范围：`(0, 1]`
```
Comment (Oreoxmt) [inline system-variables.md:1075]:
```suggestion
- 范围：`(0, 1]`，v8.0.0 及之前版本范围为 `[0, 18446744073709551615]`。
```

Comment (hi-rustin) [reply system-variables.md:1075]: We changed it in v8.1.0.

Comment (hi-rustin) [reply system-variables.md]: Thanks!

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 2]
```diff
@@ -1072,7 +1072,7 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
 - 类型：浮点数
 - 默认值：`0.5`
-- 范围：`[0, 18446744073709551615]`
+- 范围：`(0, 1]`，v8.0.0 及之前版本范围为 `[0, 18446744073709551615]`。
```
Comment (qiancai) [inline system-variables.md:1075]:
```suggestion
- 范围：`[0, 1]`，v8.0.0 及之前版本范围为 `[0, 18446744073709551615]`。
```

Comment (hi-rustin) [reply system-variables.md:1075]: 就是改到不能为0.

**Review by qiancai — APPROVED** [review]

**Comment by hi-rustin** [comment]
@Oreoxmt Can we merge this PR now?  Thanks!

**Pull request merged**